pkg/stub/handler.go

```diff
@@ -258,36 +299,90 @@ func (h *Handler) GoodConditionUpdate(srcfg *v1alpha1.SamplesResource, newStatus
-		srcfg.Name = v1alpha1.SamplesResourceName
-		srcfg.Namespace = h.namespace
-		srcfg.Kind = "SamplesResource"
-		srcfg.APIVersion = "samplesoperator.config.openshift.io/v1alpha1"
-		srcfg.Spec.Architectures = append(srcfg.Spec.Architectures, x86)
-		srcfg.Spec.InstallType = v1alpha1.CentosSamplesDistribution
-		logrus.Println("creating default SamplesResource")
-		err := h.sdkwrapper.Create(srcfg)
-		if err != nil {
-			if !kerrors.IsAlreadyExists(err) {
-				return h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "failed creating default resource: %v")
+	deleteInProgress := h.deleteInProgress
+
+	var err error
+	var srcfg *v1alpha1.SamplesResource
+	if deleteInProgress {
+		err = wait.PollImmediate(3*time.Second, 30*time.Second, func() (bool, error) {
+			sr, e := h.sdkwrapper.Get(v1alpha1.SamplesResourceName, h.namespace)
+			if kerrors.IsNotFound(e) {
+				return true, nil
 			}
-			logrus.Println("got already exists error on create default, just going to forgo this time vs. retry")
-			// just return the raw error and initiate the sdk's requeue/ratelimiter setup
-			return err
+			if err != nil && h.IsRetryableAPIError(err) {
+				return false, nil
+			}
+			if err != nil {
+				return false, err
+			}
+			// based on 4.0 testing, we've been seeing empty resources returned
+			// in the not found case, but just in case ...
+			if sr == nil {
+				return true, nil
+			}
+			// means still found ... will return wait.ErrWaitTimeout if this continues
+			return false, nil
+		})
+		if err != nil {
+			return h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "issues waiting for delete to complete: %v")
+		}
+		h.samplesResource = nil
+	} else {
+		srcfg = h.samplesResource
+	}
+	if srcfg != nil {
+		logrus.Println("SampleResource already received, not creating default")
+		return nil
+	}
+	// "4a" in the "startup" workflow, just create default
+	// resource and set up that way
+	srcfg = &v1alpha1.SamplesResource{}
+	srcfg.Name = v1alpha1.SamplesResourceName
+	srcfg.Namespace = h.namespace
+	srcfg.Kind = "SamplesResource"
+	srcfg.APIVersion = "samplesoperator.config.openshift.io/v1alpha1"
+	srcfg.Spec.Architectures = append(srcfg.Spec.Architectures, v1alpha1.X86Architecture)
+	srcfg.Spec.InstallType = v1alpha1.CentosSamplesDistribution
+	// let's reduce churn by adding conditions here vs. calling
+	// GoodConditionUpdate, so we do everything in the create vs.
+	// create then immediately update
+	condition := srcfg.Condition(v1alpha1.SamplesExist)
+	now := kapis.Now()
+	condition.LastUpdateTime = now
+	condition.Status = corev1.ConditionTrue
+	condition.LastTransitionTime = now
+	srcfg.ConditionUpdate(condition)
+	logrus.Println("creating default SamplesResource")
+	err = h.sdkwrapper.Create(srcfg)
+	if err != nil {
+		if !kerrors.IsAlreadyExists(err) {
+			return h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "failed creating default resource: %v")
```

Comment:
since we failed to create the SamplesResource, what is processError actually going to do when it tries to update a status field on SamplesResource?